lib/internal/Magento/Framework/View/Layout/Generator/Block.php

```diff
@@ -260,7 +260,7 @@ protected function getBlockInstance($block, array $arguments = [])
         }
         if (!$block instanceof \Magento\Framework\View\Element\AbstractBlock) {
             throw new \Magento\Framework\Exception\LocalizedException(
-                new \Magento\Framework\Phrase('Invalid block type: %1', [$block])
+                new \Magento\Framework\Phrase('Invalid block type: %1', [is_string($block) ? $block : get_class($block)])
```

Comment:
This code assumes that only string or object may be provided.
Please use more secure version:
`is_object($block) ? get_class($block) : $block`